src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToReferenceAProject.cs

```diff
@@ -152,5 +153,86 @@ TestProject GetTestProject(string name, string target, bool isSdkProject)
+        {
+            var tfm = "netcoreapp3.1";
+            var contentName = "script.sh";
+            var childProject = new TestProject()
+            {
+                TargetFrameworks = tfm,
+                Name = "ChildProject",
+                IsSdkProject = true
+            };
+            var childAsset = _testAssetsManager.CreateTestProject(childProject)
+                .WithProjectChanges(project => AddProjectChanges(project));
+            File.WriteAllText(Path.Combine(childAsset.Path, childProject.Name, contentName), childProject.Name);
+
+            var parentProject = new TestProject()
+            {
+                TargetFrameworks = tfm,
+                Name = "ParentProject",
+                IsSdkProject = true
+            };
+            if (copyConflictingTransitiveContent)
+            {
+                parentProject.AdditionalProperties["CopyConflictingTransitiveContent"] = "true";
+            }
+            var parentAsset = _testAssetsManager.CreateTestProject(parentProject)
+                .WithProjectChanges(project => AddProjectChanges(project, Path.Combine(childAsset.Path, childProject.Name, childProject.Name + ".csproj")));
+            File.WriteAllText(Path.Combine(parentAsset.Path, parentProject.Name, contentName), parentProject.Name);
+
+            var buildCommand = new BuildCommand(Log, Path.Combine(parentAsset.Path, parentProject.Name));
+            buildCommand.Execute().Should().Pass();
+
+            var getValuesCommand = new GetValuesCommand(Log, Path.Combine(parentAsset.Path, parentProject.Name), tfm, "ResultOutput");
+            getValuesCommand.DependsOnTargets = "Build";
+            getValuesCommand.Execute().Should().Pass();
+
+            var valuesResult = getValuesCommand.GetValuesWithMetadata().Select(pair => pair.value);
+            if (copyConflictingTransitiveContent)
+            {
+                valuesResult.Count().Should().Be(2);
+                String.Join(';', valuesResult).Should().Contain(parentProject.Name);
+                String.Join(';',valuesResult).Should().Contain(parentProject.Name);
+            }
+            else
+            {
+                valuesResult.Count().Should().Be(1);
+                valuesResult.First().Should().Contain(parentProject.Name);
+            }
+        }
+
+        private void AddProjectChanges(XDocument project, string childPath = null)
+        {
+            var ns = project.Root.Name.Namespace;
+
+            var itemGroup = new XElement(ns + "ItemGroup");
+            project.Root.Add(itemGroup);
+
+            var content = new XElement(ns + "Content",
+                new XAttribute("Include", "script.sh"), new XAttribute("CopyToOutputDirectory", "PreserveNewest"));
+            itemGroup.Add(content);
+
+            if (childPath != null)
+            {
+                var projRef = new XElement(ns + "ProjectReference",
```

Comment:
You can add a project reference via the `ReferencedProjects` property of `TestProject` instead of manipulating the XML as done here.